RadarSDK/remodel/RadarBeacon.m

```diff
@@ -0,0 +1,165 @@
+/**
+ * This file is generated using the remodel generation script.
+ * The name of the input file is RadarBeacon.value
+ */
+
+#if !__has_feature(objc_arc)
+#error This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#import "RadarBeacon.h"
+#import "RadarCollectionAdditions.h"
+
+#pragma clang diagnostic push
+#pragma GCC diagnostic ignored "-Wobjc-designated-initializers"
+
+static __unsafe_unretained NSString *const k_idKey = @"_ID";
+static __unsafe_unretained NSString *const k_descriptionKey = @"_DESCRIPTION";
+static __unsafe_unretained NSString *const kNumberMetadataKey = @"NUMBER_METADATA";
+static __unsafe_unretained NSString *const kMetadataArrayKey = @"METADATA_ARRAY";
+static __unsafe_unretained NSString *const kLocationKey = @"LOCATION";
+static __unsafe_unretained NSString *const kLocationArrayKey = @"LOCATION_ARRAY";
+static __unsafe_unretained NSString *const kVersionKey = @"VERSION";
+
+#define RMParameterAssert(condition) NSCParameterAssert((condition))
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation RadarBeacon
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    if ((self = [super init])) {
+        __id = (id)[aDecoder decodeObjectForKey:k_idKey];
+        __description = (id)[aDecoder decodeObjectForKey:k_descriptionKey];
+        _numberMetadata = (id)[aDecoder decodeObjectForKey:kNumberMetadataKey];
+        _metadataArray = (id)[aDecoder decodeObjectForKey:kMetadataArrayKey];
+        _location = (id)[aDecoder decodeObjectForKey:kLocationKey];
+        _locationArray = (id)[aDecoder decodeObjectForKey:kLocationArrayKey];
+        _version = (id)[aDecoder decodeObjectForKey:kVersionKey];
+    }
+    return self;
+}
+
+- (nullable instancetype)initWithRadarJSONObject:(nullable id)object {
+    if (!object || ![object isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+    NSDictionary *dictionary = (NSDictionary *)object;
+    if ((self = [super init])) {
+        if (!dictionary[@"_id"]) {
+            return nil;
+        }
+        __id = (NSString *)dictionary[@"_id"];
+        __description = (NSString *)dictionary[@"_description"];
```

Comment:
we could also do recursive initWithRadarJSONObject call to nested model objects within a dictionary. Just need to add category method like what we did for NSArray in `RadarCollectionAdditions.h`